duckduckgo/Android
# Remote Config (V): Https

Task/Issue URL: https://app.asana.com/0/1125189844152671/1200856548100506

### Description
This PR adds the https remote feature. It doesn't move out of the app module anything related with smarted encryption and just parses the new remote config which is then used by the `HttpsUpgrader`

### Steps to test this PR

_Config parsed correctly and working as expected_
1. Install and launch the app from this branch.
1. Check the `toggles` table in the `privacy_config` database.
1. The https toggle should be enabled.
1. Filter the logcat by `upgradable`
1. Go to `http://quora.com`
1. In the logcat you should see `quora.com is upgradable`
1. In the `toggles` table in the `privacy_config` database, change the `https` toggle to `0`.
1. Go to `http://quora.com`
1. In the logcat you should see `https is disabled in the remote config and so quora.com is not upgradable`

_Exceptions_
1. In the PR task, you will find a `privacy-config.json` file to be used during these tests.
1. Using Charles, map the call to `/trackerblocking/config/v1/android-config.json` with the json file from the task.
1. Re-install and launch the app
1. Go to `http://quora.com`
1. In the logcat you should see `quora.com is in the remote exception list and so not upgradable`
